packages-exp/auth-exp/src/core/providers/facebook.test.ts

```diff
@@ -59,7 +59,7 @@ describe('src/core/providers/facebook', () => {
     const error = AUTH_ERROR_FACTORY.create(AuthErrorCode.NEED_CONFIRMATION, {
       appName: 'foo'
     });
-    (error as TaggedWithTokenResponse)._tokenResponse = {
+    (error.customData! as TaggedWithTokenResponse)._tokenResponse = {
```

Comment:
Always having to use the ! seems weird. In storage, all errors are FirebaseStorageError which extends FirebaseError and makes customData non-optional. But I see auth-exp uses the shared ErrorFactory so I don't know if it would be really messy to get it to return an extended error type instead of FirebaseError.

Reply:
Also should a property of customData be underscore-prefixed? Not sure what that would mean.

Reply:
Valid point. What other SDKs have done is do a `is[Product]Error()` check, which narrows the type to the specific error type, then you can access `customData` safely. See [here](https://github.com/firebase/firebase-js-sdk/pull/3946/files#diff-d37d76c3ba7e573c7326dc7b8eb7ee3b7d72f6ca34b8bbc4285a64f0acfc6dcbR287-R292) as an example.

That said, I'm not too worried about them since they are all in test files. 

I think the underscore means it's for internal usage only, which is fine. @samhorlbeck